models/git/commit_status.go

```diff
@@ -283,9 +283,9 @@ func GetLatestCommitStatus(ctx context.Context, repoID int64, sha string, listOp
 		Where("repo_id = ?", repoID).And("sha = ?", sha).
 		Select("max( id ) as id").
 		GroupBy("context_hash").OrderBy("max( id ) desc")
-
-	sess = db.SetSessionPagination(sess, &listOptions)
-
+	if !listOptions.IsListAll() {
+		sess = db.SetSessionPagination(sess, &listOptions)
```

Comment:
How about put the detection into the function `SetSessionPagination`?

Reply:
I was going to do the same at first, but I referenced how it was written elsewhere.
https://github.com/go-gitea/gitea/blob/73fb1ecdcf7c72c9b15caa70a8e3f444d311bd7b/models/git/branch_list.go#L106-L115

Reply:
To avoid triggering other bugs, I've made minimal changes